bpfman-operator/controllers/bpfman-agent/common.go

```diff
@@ -109,6 +106,29 @@ func nodePredicate(nodeName string) predicate.Funcs {
 	}
 }
 
+// Predicate to watch for Pod events on a specific node which checks if the
+// event's Pod is scheduled on the given node.
+func podOnNodePredicate(nodeName string) predicate.Funcs {
```

Comment:
Andrew:
Why not just mimic what was done here https://github.com/bpfman/bpfman/blob/main/bpfman-operator/controllers/bpfman-agent/common.go#L87 Instead of making a new struct?

Andre:
I followed a different pattern which works fine too.  I don't know which is preferable, but I changed it to use the pattern we already have in nodePredicate.

Reply:
Looks good to me, thanks for keeping it consistent!